src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ReferralHelper.java

```diff
@@ -0,0 +1,352 @@
+        return documentBinaryUrl.substring(documentBinaryUrl.indexOf("/documents/"));
+    }
+
+    private static ReferralType getSelectedReferral(CaseData caseData) {
+        return caseData.getReferralCollection()
+            .get(Integer.parseInt(caseData.getSelectReferral().getValue().getCode()) - 1).getValue();
+    }
+
+    /**
+     * Creates a referral and adds it to the referral collection.
+     * @param caseData contains all the case data
+     * @param userFullName Full name of the logged-in user
+     */
+    public static void createReferral(CaseData caseData, String userFullName) {
+        if (CollectionUtils.isEmpty(caseData.getReferralCollection())) {
+            caseData.setReferralCollection(new ArrayList<>());
+        }
+
+        ReferralType referralType = new ReferralType();
+
+        referralType.setReferralNumber(String.valueOf((caseData.getReferralCollection().size() + 1)));
+        referralType.setReferCaseTo(caseData.getReferCaseTo());
+        referralType.setIsUrgent(caseData.getIsUrgent());
+        referralType.setReferralSubject(caseData.getReferralSubject());
+        referralType.setReferralSubjectSpecify(caseData.getReferralSubjectSpecify());
+        referralType.setReferralDetails(caseData.getReferralDetails());
+        referralType.setReferralDocument(caseData.getReferralDocument());
+        referralType.setReferralInstruction(caseData.getReferralInstruction());
+        referralType.setReferentEmail(caseData.getReferentEmail());
+
+        referralType.setReferralDate(Helper.getCurrentDate());
+
+        referralType.setReferredBy(userFullName);
+
+        referralType.setReferralStatus(ReferralStatus.AWAITING_INSTRUCTIONS);
+
+        referralType.setReferralHearingDate(getNearestHearingToReferral(caseData));
+
+        ReferralTypeItem referralTypeItem = new ReferralTypeItem();
+        referralTypeItem.setId(UUID.randomUUID().toString());
+        referralTypeItem.setValue(referralType);
+
+        List<ReferralTypeItem> referralCollection = caseData.getReferralCollection();
+        referralCollection.add(referralTypeItem);
+        caseData.setReferralCollection(referralCollection);
+        clearReferralDataFromCaseData(caseData);
+    }
+
+    /**
+     * Gets the next hearing date from the referral, returns "None" if no suitable hearing date exists.
+     * @param caseData contains all the case data
+     * @return Returns next hearing date in "dd MMM yyyy" format or "None"
+     */
+    private static String getNearestHearingToReferral(CaseData caseData) {
+        String earliestFutureHearingDate = HearingsHelper.getEarliestFutureHearingDate(caseData.getHearingCollection());
+
+        if (earliestFutureHearingDate == null) {
+            return "None";
+        }
+
+        Date hearingStartDate;
+        try {
+            hearingStartDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS").parse(earliestFutureHearingDate);

```

Comment:
You could do line 243 inside the try clause and then theres no need to split up the declaration and assignment of `hearingStartDate`

Reply:
Okay I didn't actually implemented this method but I do see your point as happy path is not tested here, hence I have added that hearing date to the test and can see the result is being reflected.